cmd/outline-ss-server/main.go

```diff
@@ -201,6 +218,32 @@ func (s *OutlineServer) runConfig(config Config) (func() error, error) {
 		}()
 
 		startErrCh <- func() error {
+			// Start configured web servers.
+			webServers := make(map[string]*http.ServeMux)
+			for _, srvConfig := range config.Web.Servers {
+				if _, exists := webServers[srvConfig.ID]; exists {
+					return fmt.Errorf("web server with ID `%s` already exists", srvConfig.ID)
+				}
+				mux := http.NewServeMux()
+				for _, addr := range srvConfig.Listeners {
+					server := &http.Server{Addr: addr, Handler: mux}
+					ln, err := lnSet.ListenStream(addr)
+					if err != nil {
+						return fmt.Errorf("failed to listen on %s: %w", addr, err)
+					}
+					go func() {
+						defer server.Shutdown(context.Background())
+						err := server.Serve(&HTTPStreamListener{ln})
```

Comment:
How do you stop this when the config is replaced? Is it by closing the underlying listener?
If so that's cool. The listener set is paying off. Just make sure it's working.